### DISCUSSION TITLE
System message improves chat response

### DISCUSSION THREAD

**Discussion by mogmix** [nhaouari/obsidian-textgenerator-plugin#128] [Ideas]
By adding the following to your code, the responses will improve slightly, I think. Maybe you can consider this.

```
      bodyParams["messages"] = [
        { role: "system", content: "You are ChatGPT, a large language model trained by OpenAI. Answer as concisely as possible. Current date: 14/3/2023. Knowledge cutoff: 9/1/2021." },
        { role: "user", content: params.prompt },
      ];
```

You can read it up in the OpenAI Instructions (https://platform.openai.com/docs/guides/chat/introduction). Future versions will pay even stronger attention to this.

**Comment by pi43r** [comment]
Where can you add the system param? 
gpt-3.5-turbo does not seem to work without it

Edit: Ah just needed to update the plugin. 
Still would be nice to add the system param in frontmatter or in the settings
